src/platform/forms-system/src/js/components/FormNav.jsx

```diff
@@ -111,35 +107,14 @@ export default function FormNav(props) {
       } else if (current === index) {
         setIndex(index - 1);
       }
-
-      return () => {
-        // Check main toggle to enable custom focus; the unmounting of the page
-        // before the review & submit page may cause the customScrollAndFocus
-        // function to be called inadvertently
-        if (
-          !(
-            page.chapterKey === 'review' ||
-            window.location.pathname.endsWith('review-and-submit')
-          )
-        ) {
-          if (formConfig.useCustomScrollAndFocus && page.scrollAndFocusTarget) {
-            customScrollAndFocus(page.scrollAndFocusTarget, index);
-          } else {
-            focusByOrder([defaultFocusSelector, 'h2']);
-          }
-        } else {
-          // h2 fallback for confirmation page
-          focusByOrder([defaultFocusSelector, 'h2']);
+      handleFocus(page, formConfig, index).then(() => {
+        const progressBar = document.querySelector('va-segmented-progress-bar');
+        if (progressBar) {
+          scrollTo(document.querySelector('va-segmented-progress-bar'));
         }
-      };
+      });
```

Comment:
Having this `.then()` here seems like no matter what `handleFocus` does with scrolling (it calls a `customScrollAndFocus` if it exists), the scroll will get overridden if there is a `va-segmented-progress-bar` on the page. Maybe scroll first, then call `handleFocus`?